src/rgw/rgw_op.cc

```diff
@@ -1876,6 +1878,64 @@ static void populate_with_generic_attrs(const req_state * const s,
 }
 
 
+static int filter_out_quota_info(std::map<std::string, bufferlist>& add_attrs,
+                                 const std::set<std::string>& rmattr_names,
+                                 RGWQuotaInfo& quota,
+                                 bool * quota_extracted = nullptr)
+{
+  bool extracted = false;
+
+  /* Put new limit on max objects. */
+  auto iter = add_attrs.find(RGW_ATTR_QUOTA_NOBJS);
+  std::string err;
+  if (std::end(add_attrs) != iter) {
+    quota.max_objects = (int64_t)strict_strtoll(iter->second.c_str(), 10, &err);
+    if (!err.empty()) {
+      return -EINVAL;
+    }
+    add_attrs.erase(iter);
+    extracted = true;
+  }
+
+  /* Put new limit on bucket (container) size. */
+  iter = add_attrs.find(RGW_ATTR_QUOTA_MSIZE);
+  if (iter != add_attrs.end()) {
+    const int64_t max_size =
```

Comment:
Right, it's completely unnecessary. Thanks for pointing that.
I will push the amended version soon.